tools/mpy-cross-all.py

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env python3
+import argparse
+import os
+import os.path
+
+argparser = argparse.ArgumentParser(description="Compile all .py files to .mpy recursively")
+argparser.add_argument("-o", "--out", help="output directory (default: input dir)")
+argparser.add_argument("--target", help="select MicroPython target config")
+argparser.add_argument("-mcache-lookup-bc", action="store_true", help="cache map lookups in the bytecode")
+argparser.add_argument("dir", help="input directory")
+args = argparser.parse_args()
+
+TARGET_OPTS = {
+    "unix": "-mcache-lookup-bc",
+    "baremetal": "",
+}
+
+if not args.out:
+    args.out = args.dir
+
+path_prefix_len = len(args.dir) + 1
+
+for path, subdirs, files in os.walk(args.dir):
+    for f in files:
+        if f.endswith(".py"):
+            fpath = path + "/" + f
+            #print(fpath)
+            out_fpath = args.out + "/" + fpath[path_prefix_len:-3] + ".mpy"
+            out_dir = os.path.dirname(out_fpath)
+            if not os.path.isdir(out_dir):
+                os.makedirs(out_dir)
+            cmd = "mpy-cross -v -v %s %s -o %s" % (TARGET_OPTS.get(args.target, ""), fpath, out_fpath)
```

Comment:
I think it's a good idea to use the `-s` option to `mpy-cross` to strip the prefix off fpath.  Eg if I pass in /some/long/path as the input argument to this script then I don't really want all that path taking up room in the .mpy file and then in qstrs when it gets loaded.  When you freeze a whole directory you only really care about the path/filename relative to the root of that dir.  So I suggest adding the option in the form `-s fpath[path_prefix_len:]`.  See py/mkrules.mk line 117 to see how the Makefile does a similar thing.

Note that this has nothing to do with storing qstrs multiple times.  It's just the name that is stored in the .mpy file for when exceptions are raised, to identify the file.